### ISSUE TITLE
CI hangs infinitely

### ISSUE THREAD

**Issue by shivaraj-bh** [juspay/services-flake#74]
It seems to happen while running `default-test` from [example/flake.nix](https://github.com/juspay/services-flake/blob/main/example/flake.nix). This isn’t something that happens always but whenever it does, I will have to manually restart the workflow. 

See an example workflow here: https://github.com/juspay/services-flake/actions/runs/7513430525/job/20455173179?pr=73

I can’t seem to reproduce this in my local atm.

**Comment by shivaraj-bh** [comment]
Maybe not only at `default-test`: https://github.com/juspay/services-flake/actions/runs/7513827171/job/20456036196

Either something is wrong with github actions or process-compose test process is failing to close after completion.

**Comment by shivaraj-bh** [comment]
I am suspecting it could most likely be due to port collisions while running checks parallely, as this is the only thing that is shared across processes. In this case, we might have to do something that [nixosTests](https://github.com/NixOS/nixpkgs/blob/e3528ac5517c68ef67bce66b9dce5433a402ceda/nixos/tests/kafka.nix#L80) does.

If this is the case, I should be able to reproduce locally by trying `nix flake check` multiple times but without using the `/nix/store` cache, is there a way?

**Comment by srid** [comment]
`--rebuild`?

**Comment by shivaraj-bh** [comment]
`nix flake check` doesn’t support `—rebuild`

**Comment by shivaraj-bh** [comment]
I might have stumbled upon a fix for this issue while trying to flesh out some documentation for `services-flake`. Turns out if you follow the instructions in [`Getting Started` section](https://github.com/juspay/services-flake/blob/c7bed59b874f367813966725ae390691c65cedff/README.md?plain=1#L13-L17), on the first run `pg1` is skipped because `pg1-init` fails to complete with an `initdb` error complaining about “failing to create `data` directory as it already exists". Then I scrolled up the logs in CI for a few builds that were running infinitely and [found the same error](https://github.com/juspay/services-flake/actions/runs/7558073032/job/20578768690?pr=77#step:6:4346).

Of course the `test` process depends on both `pg1` and `pg2` being up and running, and since that is not the case above, it hangs infinitely. 

**Why does it not happen all the time?**
It all depends on how the processes are spawned, it happens only when both `pg2-init` and `pg1-init` tries to create `data` directory at the same time.
